doc/howtos/software-update-server.rst

```diff
@@ -0,0 +1,281 @@
+for the file ``conf/local.conf``
+
+
+Step 0: Review and optionally modify the content of the os-core bundle
+----------------------------------------------------------------------
+The os-core bundle comes with a predefined set of components,
+however each device is likely to have slighlty different requirements.
+In some cases it might be necessary to adjust the set of components
+included in os-core.
+
+To do this, either modify or extend the ``OSTRO_IMAGE_FEATURES_REFERENCE``
+and the ``OSTRO_IMAGE_INSTALL_REFERENCE`` variables for changes
+respectively to the feature and component sets.
+
+Why modifying os-core? Because it's the only unremovable bundle and
+therefore it is well suited to deliver to the device all those tools
+that must always be available.
+
+Example::
+
+  ``OSTRO_IMAGE_INSTALL_REFERENCE += "sudo"``
+
+will add the content of the ``sudo`` recipe to the bundle os-core.
+
+.. note::
+   The step shown works only for adding components to os-core.
+
+   This is the typical case when customizing Ostro OS, because the
+   os-core configuration is very minimal.
+
+   However, in case one needs to remove something from os-core,
+   it is necessary to create either a new image recipe or a new
+   .bbappend file, where to modify the variables that define the
+   content of os-core.
+
+   These are the variable that must be modified::
+
+   - OSTRO_IMAGE_FEATURES_REFERENCE
+   - OSTRO_IMAGE_INSTALL_REFERENCE
+   - IMAGE_INSTALL
+
+   The recommended way is to modify a copy of the original recipe
+   ``ostro-image.bb`` and refer to ``ostro-image.bbclass`` for
+   the values to use.
+
+
+Step 1: Review and optionally modify the list of bundles
+--------------------------------------------------------
+Ostro already defines some bundles, and users can define additional ones.
+
+Example::
+
+  SWUPD_BUNDLES += "sudo_feature"
+  BUNDLE_CONTENTS[sudo_feature] = "sudo"
+
+This example is an alternative to add the component that was shown in
+Step 0.  Instead of extending the content of os-core, put the extra
+component in its own bundle. The main difference is that this
+approach will not install the component by default in any image.
+
+
+Step 2: Review and optionally modify the content of the images created
+----------------------------------------------------------------------
+It is also possible to choose which bundles will be pre-installed on the
+customized image, by:
+
+#. adding the bundle to the list of images generated for bundle purposes
+
+   Example::
+
+     SWUPD_IMAGES += "\
+       pre_installed_content \
+     "
+
+#. adding a varflag to SWUPD_IMAGES.
+
+   Example::
+
+     SWUPD_IMAGES[pre_installed_content]="\
+       sudo_feature \
+       add_here_other_features_if_needed \
+     "
+
+   ``SWUPD_IMAGES`` is the only token which is specific to the Ostro OS
+   syntax. The other identifiers are under the developer's control.
+
+
+This will add to the image content the ``sudo_feature`` feature from Step 1.
```

Comment:
No, the comment saying that "sudo_feature" should better be renamed to "sudo_bundle". If you rename it, here's another location where "feature" is used twice.